src/App.tsx

```diff
@@ -1,35 +1,156 @@
 import { useState } from "react";
-import { ProfilePage } from "./pages/ProfilePage/index";
+import { MyCollectionPage } from "./pages/MyCollectionPage/index";
 import { MainPage } from "./pages/MainPage/index";
+import { SignUpPage } from "./pages/SignUpPage";
+import { LoginPage } from "./pages/LoginPage";
+import { useAuth } from "./AuthContextProvider";
 
 enum Page {
   home = "Home",
-  profile = "Profile",
+  myCollection = "My Collection",
 }
 
-const allPages = Object.values(Page);
+const publicPages = [Page.home, Page.myCollection];
```

Comment:
Normally, a page like 'my collection' should not be a public page but should be a page for logged in users.